app/services/scenes/scenes.ts

```diff
@@ -199,12 +215,31 @@ export class ScenesService extends StatefulService<IScenesState> {
   @Inject() private sourcesService: SourcesService;
   @Inject() private transitionsService: TransitionsService;
 
+  protected init() {
+    // subscribe to itemAdded and itemRemoved event to sync cachedNodes
+    this.itemAdded.subscribe(itemModel => {
+      this.addItemToCache(itemModel.sceneId, itemModel.id);
+    });
```

Comment:
Do we really want to be proactively saturating the cache?  Isn't this expensive?

Reply:
I found that we a lot of calls `new SceneItem` when loading a scene-collection. We have 2 `applyProxy` method in the constructor that takes some significant time because it iterates through each property of SceneItem.

Reply:
Yeah I understand that.  Normally with a cache though you wouldn't add an object to the cache until it is being requested.  Here you are adding it as soon as it is created.  Maybe we always immediately request the object so it's effectively the same.  But the behavior struck me as weird.